### PR TITLE
add tempo ensemble results

### PR DESCRIPTION
**Pull request by idevede**
Contribution:

TEMPO with the ensemble version results have been added to this PR.

Paper: https://arxiv.org/abs/2310.04948

Github repo: https://github.com/DC-research/TEMPO 

Model Card: https://huggingface.co/Melady/TEMPO

### CODE REVIEW

**Comment by chenghaoliu89** [comment]
1. Hi @idevede could you also share the notebook about how to fine-tune your pre-trained model and evaluate on gift-eval dataset?

2. what's the difference of tempo-ensemble and the original tempo model?

**Comment by idevede** [comment]
Hi Chenghao,

Thanks for helping us update the results! Yes, we are planning to share the notebook on the evaluations. The current version's update compared to the previous TEMPO version is that:

* Probabilistic Forecasting Capability: Implemented support for probabilistic forecasting using negative log-likelihood (NLL) loss within the PyTorch Lightning framework; added capability for random prediction lengths through interpolation techniques; added the time features based on the calendar.
* 
* Optimized Ensemble Approach: Employs SLSQP optimization to determine optimal weights for ensemble models based on validation performance. The ensemble architecture includes:
    * 3 fine-tuned models trained on the complete dataset for short, medium, and long prediction horizons.
    * 15 domain-specific fine-tuned models distributed across prediction lengths (7 domains for short, 4 for medium, and 4 for long horizons).
    * 97+ dataset-specific fine-tuned models.
    
* Flexible Inference Options: Users can utilize all 115+ models collectively or selectively filter models based on their specific task requirements. For example, when forecasting for the ETT2 dataset, users can restrict model selection to the "Energy" domain to improve inference speed.

**Comment by chenghaoliu89** [comment]
Hi @idevede, thanks for sharing the technique details. It is quite an interesting idea of ensemble learning + diverse fine-tuned models !
